SwissKnife/gcloud/GCloudStorage.py

```diff
@@ -1,65 +1,79 @@
+            self.bucket_name, self.bucket_path_prefix = split_bucket(bucket)
+        elif BUCKET_NAME:
+            self.bucket_name = BUCKET_NAME
+            self.bucket_path_prefix = BUCKET_PATH_PREFIX
+        else:
+            raise RuntimeError("bucket_name param or BUCKET_PATH environment variable not available.")
 
-        self.logger = logger
-        self.logger.info(f"Building a new gcloud Storage client for bucket {BUCKET_NAME}")
         self.storage_client = gcloud.Client()
-        self.bucket = self.storage_client.get_bucket(BUCKET_NAME)
-    
-    def download_to_local_file(self, gs_path:str , local_path: str, use_bucket_path_prefix:bool=True, is_binary:bool=False):
-        """This method downloads a file from the bucket to a local file. If the 'gs_path' is complete, it will be used. But,
-        if it is only a subpath, it will be concatenated with the values BUCKET_NAME and BUCKET_PREFIX_PATH.
+        self.bucket = self.storage_client.get_bucket(self.bucket_name)
+        self.logger = logger
+        self.logger.info(f"Building a new gcloud Storage client for bucket {self.bucket_name}")
+
+    def download_to_local_file(self,
+                               gs_path: str,
+                               local_path: str,
+                               use_bucket_path_prefix: bool = True,
+                               is_binary: bool = False):
+        """This method downloads a file from the bucket to a local file. If the 'gs_path' is complete, it will be used.
+        But, if it is only a subpath, it will be concatenated with the values BUCKET_NAME and BUCKET_PREFIX_PATH.
 
         :param gs_path: The path of the Google Storage file. It can be complete or only a subpath. 
         :type gs_path: str
         :param local_path: The path where the data will be writed 
         :type local_path: str
-        :param use_bucket_path_prefix: Add the BUCKET_PREFIX_PATH to the gs_path if it is a subpath. Defaults to True
+        :param use_bucket_path_prefix: Add the BUCKET_PATH_PREFIX to the gs_path if it is a subpath. Defaults to True
         :type use_bucket_path_prefix: bool, optional
         :param is_binary: A flag that indicates if it is a binary file, defaults to False
         :type is_binary: bool, optional
         :raises RuntimeError: Raises a RuntimeError if the download process fails. 
         """
-        
-        # It is a complete path. so, it will use a custom bucket
+
         if gs_path.startswith("gs://"):
+            # It is a complete path. so, it will use a custom bucket
             gs_length = len("gs://")
             parts = gs_path[gs_length:].split("/")
             bucket_name = parts[0]
             file_path = "/".join(parts[1:])
             bucket = self.storage_client.get_bucket(bucket_name)
-        # It is not a complete path. So, it will use the BUCKET_PATH configuration
         else:
+            # It is not a complete path. So, it will use the BUCKET_PATH configuration
```

Comment:
Taking into account your changes, this will not use the `BUCKET_PATH` config, but rather the `bucket` defined in the constructor. Pointing this out to avoid confusion and check if I've understood the change correctly :P